### PR TITLE
feat: :sparkles: Miti Plasmo Extension : V2 ( Initial Version )

### PR DESCRIPTION
**Pull request by pranishpaudel**
## Release Notes

- **New Features**
  - Introduced a new GitHub Actions workflow for automated submission to the web store.
  - Added a calendar functionality with components for displaying and selecting dates.
  - Implemented a new `DateWithEvents` component to show events for selected dates.
  - Added support for Nepali weekdays in the calendar.
  - New utility functions for date manipulation and processing.
  - Introduced a month and year selection component for improved navigation.
  - Added a new `Calendar` component to enhance the user interface.
  - New `IndexPopup` component to serve as the main interface for the extension.

- **Improvements**
  - Rebranded project from "Plasmo" to "Miti extension" with updated metadata.
  - Enhanced file management with improved `.gitignore` settings.
  - Introduced new configuration files for Tailwind CSS and PostCSS for styling.

- **Bug Fixes**
  - Addressed issues with calendar data processing and display.

### CODE REVIEW

**Comment by Parajulibkrm** [comment]
@nirajacharyaa could you check this?

**Comment by nirajacharyaa** [comment]
> @nirajacharyaa could you check this?

okay

File: apps/plasmo/src/components/Calendar/CalendarBody.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+const CalendarBody: React.FC = () => {
+  const [calendarData, setCalendarData] = useState<ProcessedDay[][]>([])
+  const [selectedDay, setSelectedDay] = useState<ProcessedDay | null>(null)
+  const [todayBSDay, setTodayBSDay] = useState<string | null>(null)
+
+  useEffect(() => {
+    const fetchCalendarData = async () => {
+      try {
+        const response = await fetch(MITI_API_ROUTE("2081", "07"))
+        const data: CalendarDayData[] = await response.json()
```
Comment (nirajacharyaa) [inline apps/plasmo/src/components/Calendar/CalendarBody.tsx]: you are fetching calendar data manually, you should fetch data of the current month and give user privilege to switch months with some UI

you need to use nepali-datetime npm package

File: apps/plasmo/src/components/Calendar/helpers/dateUtils.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  return selectedNum - todayNum
+}
+
+export const getRelativeDayText = (dayDiff: number): string => {
+  if (dayDiff === 0) return "today"
+  if (dayDiff === 1) return "tomorrow"
+  if (dayDiff === -1) return "yesterday"
+  if (dayDiff > 1) return `${dayDiff} days after`
+  return `${Math.abs(dayDiff)} days before`
+}
```
Comment (nirajacharyaa) [inline apps/plasmo/src/components/Calendar/helpers/dateUtils.ts]: this thing can be replaced by the date-fns package

File: apps/plasmo/src/components/Calendar/types/types.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    }
+  }
+  eventDetails: Array<{
+    isHoliday: boolean
+    title: {
+      np: string
+      en: string | null
+    }
+  }>
+}
```
Comment (nirajacharyaa) [inline apps/plasmo/src/components/Calendar/types/types.ts:35]: the repo has internal type package, the types can be used from there

File: apps/plasmo/src/components/Calendar/helpers/dateUtils.ts [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    today.getMonth(),
+    parseInt(selectedDay.date || "1")
+  )
+
+  return {
+    dayName: dateObj.toLocaleString("en-US", { weekday: "long" }),
+    monthName: dateObj.toLocaleString("en-US", { month: "long" }),
+    year: today.getFullYear()
+  }
+}
```
Comment (nirajacharyaa) [inline apps/plasmo/src/components/Calendar/helpers/dateUtils.ts]: these tooo

**Review by nirajacharyaa — CHANGES_REQUESTED** [review]

File: apps/plasmo/src/components/Calendar/helpers/dateUtils.ts [hunk 5]
```diff
@@ -1,5 +1,33 @@
...
+// Convert Nepali year to English year
+const getEnglishYear = (nepaliYear: number, nepaliMonth: string): number => {
+  // Base conversion (approximate)
+  const baseEnglishYear = nepaliYear - 57
+
+  // Adjust for months that span into the next year
+  const monthData = nepaliToEnglishMonthMap[nepaliMonth]
+  return baseEnglishYear + (monthData?.year || 0)
+}
+
```
Comment (nirajacharyaa) [inline apps/plasmo/src/components/Calendar/helpers/dateUtils.ts]: @pranishpaudel  please use [nepali-datetime](https://www.npmjs.com/package/nepali-datetime) npm package, the English month change date is not even valid for each year, it varies.

**Comment by pranishpaudel** [comment]
Sure

**Comment by pranishpaudel** [comment]
Please check the latest commit and tell me if any updates are needed in it

**Comment by Parajulibkrm** [comment]
Looks good for a starting point. Thanks @pranishpaudel

**Pull request merged**